custom/enikshay/tasks.py

```diff
@@ -301,39 +245,44 @@ def get_adherence_schedule_start_date(self):
             return None
 
     @staticmethod
-    def count_doses_taken(dose_taken_by_date, start_date=None, end_date=None):
+    def count_doses_taken(dose_status_by_date, start_date=None, end_date=None):
         """
         Args:
-            dose_taken_by_date: result of self.calculate_doses_taken_by_day
+            dose_status_by_date: result of self.calculate_dose_status_by_day
         Returns:
             total count of adherence_cases excluding duplicates on a given day. If there are
             two adherence_cases on one day at different time, it will be counted as one
         """
+        return EpisodeAdherenceUpdate.count_doses_of_status('taken', dose_status_by_date, start_date, end_date)
+
+    @staticmethod
+    def count_doses_of_status(status, dose_status_by_date, start_date=None, end_date=None):
+        """status should be 'taken', 'unknown', or 'missed'"""
         if bool(start_date) != bool(end_date):
             raise EnikshayTaskException("Both of start_date and end_date should be specified or niether of them")
 
         if not start_date:
-            return len([dose_taken for dose_taken in dose_taken_by_date.values() if dose_taken])
+            return len([ds for ds in dose_status_by_date.values() if getattr(ds, status)])
```

Comment:
i am confused about what `getattr(ds, status)` should be? isn't `ds` an instance of the `namedtuple`?